js_modules/dagit/packages/core/src/workspace/WorkspaceContext.tsx

```diff
@@ -162,21 +176,69 @@ const useWorkspaceState = () => {
       return [...accum, ...reposForLocation];
     }, [] as DagsterRepoOption[]);
 
-    return {error: null, options};
+    return {error: null, allRepos};
   }, [workspaceOrError]);
 
+  const [visibleRepos, toggleVisible] = useVisibleRepos(allRepos);
+
   return {
     refetch,
     loading: loading && !data, // Only "loading" on initial load.
     error,
     locationEntries,
-    allRepos: options,
+    allRepos,
+    visibleRepos,
+    toggleVisible,
   };
 };
 
+/**
```

Comment:
These changes are just moving the "visible repos" concept out of the left sidebar up into the global workspace context so that the same visibility state can be shared across the app.